src/pymatgen/io/abinit/pseudos.py

```diff
@@ -601,7 +601,9 @@ def _dict_from_lines(lines, key_nums, sep=None) -> dict:
     if len(lines) != len(key_nums):
         raise ValueError(f"{lines = }\n{key_nums = }")
 
-    kwargs = Namespace()
```

Comment:
The original implementation uses the `Namespace` dict, which according to the original docstring (before my clean up in https://github.com/materialsvirtuallab/monty/pull/722):
> A dictionary that does not permit to redefine its keys.

The issue is two-fold:
1. My fault goes first, looks like the override on `update` is not complete to handle `Iterator` correctly, and surely I would fix this ASAP:
```python
from monty.collections import Namespace

namespace_dict = Namespace()

namespace_dict.update(zip(["a", "b"], [1, 2]))
print(namespace_dict)

namespace_dict.update({"a":1, "b":2})
print(namespace_dict)
``` 

After fixing the issue, however the usage of `Namespace` in `oncvpsp_header` method of `io.abinit.pseudo.NcAbinitHeader` seems incorrect with `pop`:
https://github.com/materialsproject/pymatgen/blob/b28b4c24d92830327f2aaf199804b726d680437c/src/pymatgen/io/abinit/pseudos.py#L783